tests/test_open_images_format.py

```diff
@@ -8,12 +8,82 @@
+from datumaro.plugins.open_images_format import OpenImagesConverter, OpenImagesImporter
+from datumaro.util.image import Image
+from datumaro.util.test_utils import TestDir, compare_datasets_strict
 from datumaro.components.extractor import AnnotationType, DatasetItem, Label, LabelCategories
 from datumaro.components.dataset import Dataset
 from tests.requirements import Requirements, mark_requirement
 
+class OpenImagesFormatTest(TestCase):
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_can_save_and_load(self):
+        source_dataset = Dataset.from_iterable(
+            [
+                DatasetItem(id='a', subset='train',
+                    annotations=[Label(0, attributes={'score': 0.7})]
+                ),
+                DatasetItem(id='b', subset='train', image=np.zeros((8, 8, 3)),
+                    annotations=[Label(1), Label(2, attributes={'score': 0})]
+                ),
+            ],
+            categories={
+                AnnotationType.label: LabelCategories.from_iterable([
+                    '/m/0',
+                    ('/m/1', '/m/0'),
+                    '/m/2',
+                ]),
+            },
+        )
+
+        with TestDir() as test_dir:
+            OpenImagesConverter.convert(source_dataset, test_dir,
+                save_images=True)
+
+            parsed_dataset = Dataset.import_from(test_dir, 'open_images')
+
+            # the converter assumes that labels without a score have a score of 100%
+            source_dataset.get('b', subset='train').annotations[0].attributes['score'] = 1
+
+            compare_datasets_strict(self, source_dataset, parsed_dataset)
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_can_save_and_load_with_no_subsets(self):
+        source_dataset = Dataset.from_iterable(
+            [
+                DatasetItem(id='a',
+                    annotations=[Label(0, attributes={'score': 0.7})]
+                ),
+            ],
+            categories={
+                AnnotationType.label: LabelCategories.from_iterable(['/m/0']),
+            },
+        )
+
+        with TestDir() as test_dir:
+            OpenImagesConverter.convert(source_dataset, test_dir,
+                save_images=True)
+
+            parsed_dataset = Dataset.import_from(test_dir, 'open_images')
+
+            compare_datasets_strict(self, source_dataset, parsed_dataset)
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_can_save_and_load_image_with_arbitrary_extension(self):
+        dataset = Dataset.from_iterable([
+            DatasetItem(id='a/1', image=Image(path='a/1.JPEG',
+                data=np.zeros((4, 3, 3)))),
+            DatasetItem(id='b/c/d/2', image=Image(path='b/c/d/2.bmp',
+                data=np.zeros((3, 4, 3)))),
+        ], categories=[])
+
+        with TestDir() as test_dir:
+            OpenImagesConverter.convert(dataset, test_dir, save_images=True)
+            #import pdb; pdb.set_trace()
```

Comment:
Please, remove the comment.
Probably, you'd want to set up a debugger in VS Code / your favourite IDE.